src/null-safety/index.md

```diff
@@ -86,10 +76,56 @@ Dart null safety support is based on the following three core design principles:
   fewer bugs, but smaller binaries and faster execution.
 
 
-## Enabling/disabling null safety {#enable-null-safety}
+## Dart 3 and null safety
+
+Dart 3---planned for a mid-2023 release---requires sound null safety. It
+will prevent code from running without it.
+
+Packages without null safety support will cause issues when resolving
+dependencies:
+
+```
+$ dart pub get
+
+Because mypackage doesn't support null safety, version solving failed.
+The lower bound of '>=2.9.0 <3.0.0' must be 2.12.0 or higher to enable null safety.
+```
+
+Libraries that opt out of null safety will cause analysis or compilation
+errors:
+
+```
+$ dart run bin/hi.dart
+
+../testpkg1/lib/testpkg1.dart:1:1: Error: Library doesn't support null safety.
+// @dart=2.9
+^^^^^^^^^^^^
+```
+
+To resolve these issues, you must [migrate](#migrate) all existing code to use
+sound null safety.
+
+### Dart 3 backwards compatibility
+
+Packages and apps migrated to use null safety with Dart 2.12 or later will
+likely be backwards compatible with Dart 3. Specifically, for any package where
+the lower bound of the SDK constraint is 2.12.0 or higher, pub will allow
+resolution even when the upper bound is limited to versions below 3.0.0. For
+example, a package with the following constraint will be allowed to resolve with
+a Dart 3.x SDK:
+
+```yaml
+environment:
+  sdk: '>=2.14.0 <3.0.0'
+```
+
+This allows developers to use Dart 3 sound null safety with packages that have
+been migrated to 2.12 null safety without needing a second migration.
+
+## Dart 2.x and null safety {#enable-null-safety}
 
-You can use sound null safety in Dart 2.12 and Flutter 2.0 or later.
-Dart 3 and later will [_only_ support sound null safety][Dart 3 sound null safety tracking issue].
+In Dart 2.12 or later, null safety is a configuration option in the pubspec.
```

Comment:
Maybe be extra clear that this is only pre Dart 3?  "In Dart 2.12 or later (but pre-Dart 3.0)"

Reply:
Or better "In Dart SDK versions 2.12 to 2.19, ..."